server.js

```diff
@@ -13,8 +14,70 @@ app.get("/", function (request, response) {
   response.send("Hotel booking server.  Ask for /bookings, etc.");
 });
 
-// TODO add your routes and helper functions here
+// return all bookings /bookings
 
-const listener = app.listen(process.env.PORT, function () {
-  console.log("Your app is listening on port " + listener.address().port);
+app.get("/bookings", (req, res) => {
+  res.send(bookings)
+});
+
+// return bookings by search term
+
+app.get("/bookings/search", (req, res) => {
+  let searchTerm = req.query.term.toLowerCase()
+ 
+  let searchBookings = bookings.filter((booking) => booking.firstName.toLowerCase().includes(searchTerm)|| 
+  booking.surname.toLowerCase().includes(searchTerm) || 
+  booking.email.toLowerCase().includes(searchTerm))
+
+  if(searchBookings.length > 0){
+    res.status(200).json(searchBookings)
+  } else{
+    res.send({message: 'no bookings found'})
+  }
+});
+
+// return one booking /bookings/:id
+
+app.get("/bookings/:id", (req,res) => {
+  const id = Number(req.params.id);
+  const getbooking = bookings.find(booking => booking.id === id);
+  res.status(200).json(getbooking)
+})
+
+
+let maxID = Math.max(...bookings.map(c => c.id));
+
+// post a booking
+app.post("/bookings", (req,res) => {
```

Comment:
I would recommend looking at this function. I created a new booking and was presented with this as the output

`  {
    "id": 6,
    "body": {
      "title": "Mr",
      "firstName": "Louis",
      "surname": "Boyd",
      "email": "jimi@example.com",
      "roomId": 21,
      "checkInDate": "2017-11-21",
      "checkOutDate": "2017-11-23"
    }
  }`